hedgehog/src/Hedgehog/Internal/Runner.hs

```diff
@@ -245,25 +225,33 @@ checkNamed region mcolor name prop = do
 -- | Check a property.
 --
 check :: MonadIO m => Property -> m Bool
-check prop =
+check prop = do
+  mcolor <- detectColor
   liftIO . displayRegion $ \region ->
-    (== OK) . reportStatus <$> checkNamed region Nothing Nothing prop
+    (== OK) . reportStatus <$> checkNamed region mcolor Nothing prop
 
 -- | Check a property using a specific size and seed.
 --
 recheck :: MonadIO m => Size -> Seed -> Property -> m ()
 recheck size seed prop0 = do
+  mcolor <- detectColor
   let prop = withTests 1 prop0
   _ <- liftIO . displayRegion $ \region ->
-    checkRegion region Nothing Nothing size seed prop
+    checkRegion region mcolor Nothing size seed prop
   pure ()
 
+data Parallelism =
+    UseParallelism
+  | NoParallelism
+
 -- | Check a group of properties using the specified runner config.
 --
-checkGroup :: MonadIO m => RunnerConfig -> Group -> m Bool
-checkGroup config (Group group props) =
+checkGroup :: MonadIO m => Parallelism -> Group -> m Bool
```

Comment:
What about all the other options we had when passing a `RunnerConfig`? We were able to specify color, verbosity, workers, (even a fixed seed as discussed in https://github.com/hedgehogqa/haskell-hedgehog/issues/201#issuecomment-441401915).

Could you please explain the motivation behind this breaking change?

Reply:
Fixed seed wasn't part of the `RunnerConfig` I removed -- only verbosity, color and workers.

The first version I wrote (privately, never pushed here) actually did just removed the `Maybe` part of the `RunnerConfig` and forced choice early on all three fields... until I noticed that only color was every checked in a tight loop, the other weren't. Workers doesn't make sense to check after starting to run properties anyway, and verbosity seemed (at a quick glance and lack of compiler complaints from removing it altogether) like an artifact from before progress formatters were implemented. I didn't check in depth, but it doesn't actually seem to do anything at all.

In the end, I noticed neither workers nor (it seemed) verbosity were used after running properties had started, so I just gutted everything to make that fact more explicit. The change doesn't actually break anything, terminals can't change color support halfway through (unless you really enjoy changing your term's settings while hedgehog is running) and the other two weren't actually changing anything after property checking started.

It could make sense to keep `RunnerConfig` around, I don't feel strongly. I just guessed it was an artifact from the Ye Olde Days based on how it was (not) used. Lemme know.

Reply:
:+1: Having only the minimum required changes in this PR will increase the chances of getting it merged sooner, at least until @jystic is back. We can always tweak/remove `RunnerConfig` separately.